src/main/java/server/ServerHandler.java

```diff
@@ -30,13 +33,43 @@
     private final Set<Room> rooms;
     private final Validator validator;
     private final Room DEFAULT_ROOM = new Room("DEFAULT");
+    private final static Function<String, Command> DEFAULT_COMMAND = cmds -> new CommandDefault();
+    private final Map<String, Function<String, Command>> commandsFactory;
 
     public ServerHandler(Viewer viewer) {
         this.viewer = Objects.requireNonNull(viewer);
         sessionSet = new HashSet<>();
         rooms = new HashSet<>();
         rooms.add(DEFAULT_ROOM);
         validator = new Validator(viewer);
+
+        commandsFactory = Map.of(
+                "/list", cmds -> cmds.split(" ").length == 1 ? new CommandList(sessionSet) : new CommandErrorSyntax(),
+                "/help", cmds -> cmds.split(" ").length == 1 ? new CommandHelp() : new CommandErrorSyntax(),
+                "/rooms", cmds -> cmds.split(" ").length == 1 ? new CommandRooms(rooms) : new CommandErrorSyntax(),
```

Comment:
There is a lot of common. Probably, we should choose a different type for the argument we pass to the factory method